core/eth/state.go

```diff
@@ -28,8 +45,12 @@ func (cs *ChainState) GetOperatorStateByOperator(ctx context.Context, blockNumbe
 		return nil, err
 	}
 
-	return getOperatorState(operatorsByQuorum, uint32(blockNumber))
+	err = cs.refreshSocketMap(ctx, operatorsByQuorum)
+	if err != nil {
+		return nil, err
+	}
 
+	return getOperatorState(operatorsByQuorum, uint32(blockNumber), cs.SocketMap)
```

Comment:
instead of `getOperatorState` being a static function and passing in cs.SocketMap, why not turn it into a method, which has access to `cs` members?